src/main/flight/mixer.c

```diff
@@ -743,27 +748,125 @@ STATIC_UNIT_TESTED void servoMixer(void)
+                throttle3dPosition = THROTTLE_3D_POSITIVE;
+            } else {
+                // when starting in neutral go positive and  when coming from positive. Keep positive throttle within deadband
+                if ((throttle3dPosition == THROTTLE_3D_NEUTRAL && !throttleMinPrevious)
+                    || (throttle3dPosition == THROTTLE_3D_NEUTRAL && (throttleMinPrevious >= flight3DConfig->deadband3d_high))) {
+                    throttleMax = escAndServoConfig->maxthrottle;
+                    throttle = throttleMin = flight3DConfig->deadband3d_high;
+                // When coming from negative. Keep negative throttle within deadband
+                } else if (throttle3dPosition == THROTTLE_3D_NEUTRAL && (throttleMinPrevious <= flight3DConfig->deadband3d_low)) {
+                    throttleMax = escAndServoConfig->mincommand;
+                    throttle = throttleMin = flight3DConfig->deadband3d_low;
+                }  else {
+                    throttleMax = throttleMaxPrevious;
+                    throttleMin = throttleMinPrevious;
+                    throttle = throttlePrevious;
+                }
+            }
 
-        bool isFailsafeActive = failsafeIsActive();
+            throttleMaxPrevious = throttleMax;
+            throttleMinPrevious = throttleMin;
+            throttlePrevious = throttle;
+        } else {
+            throttleMin = escAndServoConfig->minthrottle;
+            throttleMax = escAndServoConfig->maxthrottle;
+        }
+
+        throttleRange = throttleMax - throttleMin;
+
+        if (rollPitchYawMixRange > throttleRange) {
+            motorLimitReached = true;
+            for (i = 0; i < motorCount; i++) {
+                rollPitchYawMix[i] = (rollPitchYawMix[i] * throttleRange) / rollPitchYawMixRange;
+
+                // Get the max correction from center when agressivity enabled. (Some setups don't like this option)
+                if (mixerConfig->agressive_airmode) throttleMin = throttleMax = throttleMin + (throttleRange / 2);
+            }
+        } else {
+            motorLimitReached = false;
+            throttleMin = throttleMin + (rollPitchYawMixRange / 2);
+            throttleMax = throttleMax - (rollPitchYawMixRange / 2);
+        }
+
+        // Now add in the desired throttle, but keep in a range that doesn't clip adjusted
+        // roll/pitch/yaw. This could move throttle down, but also up for those low throttle flips.
+        for (i = 0; i < motorCount; i++) {
+            motor[i] = rollPitchYawMix[i] + constrain(throttle * currentMixer[i].throttle, throttleMin, throttleMax);
+
+            if (isFailsafeActive) {
+                mixConstrainMotorForFailsafeCondition(i);
+            } else if (feature(FEATURE_3D)) {
+                if (throttle3dPosition == THROTTLE_3D_NEGATIVE) {
+                    motor[i] = constrain(motor[i], escAndServoConfig->minthrottle, flight3DConfig->deadband3d_low);
```

Comment:
It's not that hard to read the code ;-)
That why I complained about I-term limit macro - it did take quite a while to understand ..